pages/5_📈_Validator_Scores_Monitor.py

```diff
@@ -0,0 +1,137 @@
+import requests
+import streamlit as st
+from streamlit_timeline import st_timeline
+import time, datetime
+import json, requests
+import plotly.graph_objs as go
+from datetime import timedelta
+
+
+# For test
+# URL_GET_MINER_TIMELINE = "http://20.62.195.114:8000/get_miner_timeline"
+# URL_GET_MINER_INFO = "http://20.62.195.114:8000/get_miner_info"
+# VALID_UIDS = [str(x) for x in range(256)]
+# DEFAULT_VALIDATOR_UID = 1
+
+# For production
+URL_GET_MINER_TIMELINE = "http://nichestorage.nichetensor.com:10000/get_miner_timeline"
+URL_GET_MINER_INFO = "http://nichestorage.nichetensor.com:10000/get_miner_info"
+VALID_UIDS = ["0", "202", "232", "242", "133", "132", "228", "28", "121"]
+DEFAULT_VALIDATOR_UID = 202
+
+LST_MODEL_NAMES = ["GoJourney", "AnimeV3", "JuggernautXL", "RealitiesEdgeXL", "DreamShaperXL", "Gemma7b", "Llama3_70b","FaceToMany", "StickerMaker", "DallE"]
+st.set_page_config( layout="wide")
+
+timeline_miner_info, validator_data = {}, []
+
+
+def get_miner_timeline(validator_id, miner_id):
+
+    json_data = {
+        'validator_uid': int(validator_id),
+        'miner_uid': int(miner_id),
+    }
+
+    response = requests.post(URL_GET_MINER_TIMELINE, json=json_data)
+    result = response.json()
+    return result
+
+def plot_timeline_chart(dates, values, miner_data):
+    if len(values) > 0:
+        scores = [float(x) for x in values]
+        fig = go.Figure()
+
+        fig.add_trace(go.Scatter(x=dates, y=scores, mode='lines+markers', name='Scores'))
+
+        fig.update_layout(
+            title=f"Timeline scores of miner {miner_data['uid']} (model {miner_data.get('model_name','')})",
+            xaxis_title='Date',
+            yaxis_title='Score',
+            xaxis=dict(
+                tickformat='%H:%M:%S',
+                tickangle=-45,
+                # tickvals=[date + timedelta(minutes=10) for date in dates]
+            ),
+            yaxis=dict(
+                range=[min(scores) - 0.01, max(scores) + 0.01]
+            ),
+            width=1200
+        )
+
+        st.plotly_chart(fig)
+    
+
+def get_miner_info_timeline():
+    global timeline_miner_info
+    try:
+        response = requests.get(URL_GET_MINER_INFO)
+        data = response.json()
+        timeline_miner_info = data
+        st.session_state.timeline_miner_info = timeline_miner_info
+    except Exception as ex:
+        print("[ERROR] Fetch data: ", str(ex))
+        
+get_miner_info_timeline()
```

Comment:
Check if timeline_miner_info already existed in session_state, don't need to do get
if not, get timeline_miner_info from api

This checking avoid getting from api every when changing any streamlit variable (eg. validator_select, model_select, miner_select)